src/uct/cuda/cuda_ipc/cuda_ipc_cache.h

```diff
@@ -16,36 +16,36 @@
 #include <cuda_runtime.h>
 
 
-typedef struct uct_cuda_ipc_cache        uct_cuda_ipc_cache_t;
-typedef struct uct_cuda_ipc_cache_region uct_cuda_ipc_cache_region_t;
 typedef struct uct_cuda_ipc_rem_memh     uct_cuda_ipc_rem_memh_t;
 
 
-struct uct_cuda_ipc_cache_region {
-    ucs_pgt_region_t        super;        /**< Base class - page table region */
-    ucs_list_link_t         list;         /**< List element */
-    uct_cuda_ipc_key_t      key;          /**< Remote memory key */
-    void                    *mapped_addr; /**< Local mapped address */
-    uint64_t                refcount;     /**< Track inflight ops before unmapping*/
-};
-
-
-struct uct_cuda_ipc_cache {
-    pthread_rwlock_t      lock;       /**< protests the page table */
-    ucs_pgtable_t         pgtable;    /**< Page table to hold the regions */
-    char                  *name;      /**< Name */
-};
+/**
+ * Structure associated with rcache region. ipc_handle is returned along with
+ * the start of the mapping to check if VA recycling has occurred.
+ */
+typedef struct uct_cuda_ipc_rcache_region {
+    ucs_rcache_region_t     super;
+    CUipcMemHandle          ipc_handle;
+    void                    *mapping_start;
+} uct_cuda_ipc_rcache_region_t;
 
 
-ucs_status_t uct_cuda_ipc_create_cache(uct_cuda_ipc_cache_t **cache,
-                                       const char *name);
+ucs_status_t
+uct_cuda_ipc_create_cache(uct_cuda_ipc_md_t *md, ucs_rcache_t **cache,
+                          const char *name);
```

Comment:
This method can be removed from header file. Because it is only used inside `uct_cuda_ipc_map_memhandle`.